File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let map = new Map(); // value -> index

    for (let i = 0; i < nums.length; i++) {
        let complement = target - nums[i];
        if (map.has(complement)) {
            let idx1 = map.get(complement);
            let idx2 = i;
            if (idx1 > idx2) [idx1, idx2] = [idx2, idx1]; // ensure increasing order
            return [idx1, idx2];
        }
        map.set(nums[i], i); // store index of current value
    }
    
    return [];
};

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>mpp;
        for(int i=0; i<nums.size(); i++){
            int comp=target-nums[i];
            if(mpp.find(comp)!=mpp.end()){
                int idx1=mpp[comp];
                int idx2=i;
                if(idx1 > idx2){
                   swap(idx1,idx2);
                }
                return {idx1,idx2};
            }
            mpp[nums[i]]=i;
            
        }
        return {};
        
    }
};